libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -433,6 +437,54 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
 
 }
 
+#include "AP_DDS_Service_Table.h"
+
+void AP_DDS_Client::on_request([[maybe_unused]] uxrSession* uxr_session, uxrObjectId object_id, [[maybe_unused]] uint16_t request_id, SampleIdentity* sample_id, ucdrBuffer* ub, [[maybe_unused]] uint16_t length, void* args){
+    switch (object_id.id){
+        case services[to_underlying(ServiceIndex::ARMING_MOTORS)].rep_id:
+        bool arm;
+        const bool deserialize_success = ucdr_deserialize_bool(ub,&arm);
+        if(deserialize_success == false){
+            break;
+        }
+        
+        if(arm == true) {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Request for arming recieved");
+        } else {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Request for disarming recieved");
+        }
+        
+        const uxrObjectId replier_id = {
+            .id = services[to_underlying(ServiceIndex::ARMING_MOTORS)].rep_id,
+            .type = UXR_REPLIER_ID
+        };
+        uint8_t reply_buffer[8] = {
+            0
+        };
+        ucdrBuffer reply_ub;
+
+        // dummy code
+        // TODO : add arming/disarming checks here 
+        bool result = arm;
+        ucdr_init_buffer(&reply_ub, reply_buffer, sizeof(reply_buffer));
+        const bool serialize_success = ucdr_serialize_bool(&reply_ub,result);
+        if(serialize_success == false){
+            break;
+        }
+
+        uint32_t* count_ptr = (uint32_t*) args;
+        (*count_ptr)++;
+
+        uxr_buffer_reply(uxr_session, reliable_out, replier_id, sample_id, reply_buffer, sizeof(reply_buffer));
```

Comment:
```suggestion
        uxr_buffer_reply(uxr_session, reliable_out, replier_id, sample_id, reply_buffer, sizeof(result));
```

Correcting my earlier suggestion. The last arg should be the size of the serialised data, not the buffer size.

Reply:
Thanks a lot @srmainwaring ! Have made the requested changes